packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,186 @@
+# Banner
+
+## Overview
+
+The banner is a component used to display a persistent notification to a user.
+
+### Background
+
+[Design specification](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/29c405f7-08ea-48b6-973f-546970b9dbab/)
+
+[GitHub issue](https://github.com/ni/nimble/issues/305)
+
+### Non-goals
+
+- Arbitrary HTML content
+- A container component that would enforce design constraints on multiple banners displayed together
+- Limitating length of displayed text and/or truncating text with ellipsis
+
+### Features
+
+- Optionally dismissable
+- Optional link or button (only 1) to perform an action/navigation
+- Distinct designs for banners of type Error, Warning, Info, or Neutral
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+Refer to examples in the "Some examples of screenshots" section of the [GitHub issue](https://github.com/ni/nimble/issues/305).
+
+---
+
+## Design
+
+The banner is a standalone component that spans the width of its containing element, presenting, from left to right, an icon (dependent on Error/Warning/Info type), an optional title, an optional plaintext message, an optional action button or link, and an optional close control. The banner is intended to display short messages (recommended three lines or fewer) that communicate some persistent state or condition.
+
+In this initial implementation, we will not limit the height of the banner. It will grow to fit the text that it is given.
+
+When the user presses the close button, we will hide the banner (and fire a `close` event).
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*: `nimble-banner`
+- *Props/Attrs*:
+  - `heading` - if set, a heading with the specified text is shown to the left of the message. OPEN QUESTION: should we instead use the existing `title` attribute?
+  - `text` - the message text
+  - `prevent-dismiss` - set to hide the close button (attr name taken from Nimble dialog)
+  - `type` - one of `error`, `warning`, `info`, or `neutral` (default)
```

Comment:
I had avoided that in case we later decide to implement the second version of banners that were originally in Brandon's design doc (i.e. the ones with just an outline rather than full background color). He had referred to those as low vs high severity. But the point about the icons and tooltip are convincing. I'll make the change.

Reply:
I think if we needed to add outline vs filled background in the future, that would align better with our `appearance` attributes that some components have.